utility/remove_unused.py

```diff
@@ -136,97 +126,109 @@ def main():
-                )
-
-            to_remove = []
-            last_six_months = {}
-            last_two_months = {}
-
-            today = dt.now()
-            for backup in list_of_backups:
-                backup_dt = dt.strptime(backup[: backup.index('.')], backup_date_format)
-                month_difference = diff_month(today, backup_dt)
-                if month_difference > 6:
-                    to_remove.append(backup)
-                elif month_difference > 2:
-                    month = backup_dt.month
-                    if month in last_six_months:
-                        if last_six_months[month] > backup:
-                            to_remove.append(backup)
-                        else:
-                            to_remove.append(last_six_months[month])
-                            last_six_months[month] = backup
-                    else:
-                        last_six_months[month] = backup
-                else:
-                    backup_path = os.path.join(backup_directory, backup)
-                    currently_processed_backup_hash = hash_node(backup_path)
-                    if currently_processed_backup_hash in last_two_months:
-                        if last_two_months[currently_processed_backup_hash] > backup:
-                            to_remove.append(backup)
-                        else:
-                            to_remove.append(last_two_months[currently_processed_backup_hash])
-                    last_two_months[currently_processed_backup_hash] = backup
-            for backup in to_remove:
-                backup_path = os.path.join(backup_directory, backup)
-                if backup_path != backup_name_latest:
-                    if os.path.isdir(backup_path):
-                        shutil.rmtree(backup_path)
-                    elif os.path.isfile(backup_path):
-                        os.unlink(backup_path)
-
         logger.info('Removing old cache json files')
-        remove_old_backups('confd')
+        remove_old_backups(os.path.join(cache_directory, 'confd'))
     except Exception as e:
         logger.exception('Exception found while running remove_unused script')
         raise e
     logger.info('Job finished successfully')
 
 
+def represents_int(s: str) -> bool:
+    try:
+        int(s)
+        return True
+    except ValueError:
+        return False
+
+
+def represents_uuid(s: str) -> bool:
+    try:
+        uuid.UUID(s)
+        return True
+    except ValueError:
+        return False
+
+
+def remove_old_backups(backup_directory: str):
+    """
+    Deduplicate backups and keep the latest copy only. Keep one backup per month for the last 6 months.
+    For the last 2 month, keep all unique backups. All other files are removed.
+
+    Arguments:
+        :param backup_directory  (str) Path to directory to search for backups.
+    """
+    list_of_backups = get_list_of_backups(backup_directory)
```

Comment:
In `get_list_of_backups` please handle the case if directory does not exists.
It will fail on `os.listdir(dir_name)` now